internal/blockprocessor/processor.go

```diff
@@ -56,6 +56,20 @@ func New(conf *Config) *BlockProcessor {
 	}
 }
 
+// Bootstrap initializes the ledger and database with the first block, which contains a config transaction.
+// This block is a.k.a. the "genesis block".
+func (b *BlockProcessor) Bootstrap(configBlock *types.Block) error {
+	if err := b.recoverWorldStateDBIfNeeded(); err != nil {
+		panic(errors.WithMessage(err, "error while recovering node"))
+	}
```

Comment:
I am not sure whether this is needed. We would reach here only if the block height is 0. When it is 0, all other stores must be empty.

Reply:
You have a point here... I added it because there some failures during tests, probably for the init of the trie... so I added both. Looking into the recover shows it makes no senses because indeed the ledger height and store height will bothe be 0.  

Nevertheless, looking deeper, I think I did not pay enough attention to the difference between:
```
	blockStoreHeight, err := b.blockStore.Height()
	stateDBHeight, err := b.committer.db.Height()
```
and when calling the `Bootstrap()` method in the `transaction_processor.go` I get the store height and not the ledger height:
```

	ledgerHeight, err := conf.db.Height()
	if err != nil {
		return nil, err
	}
	if ledgerHeight == 0 {
		p.logger.Info("Bootstrapping the ledger and database")
		tx, err := PrepareBootstrapConfigTx(conf.config)
		if err != nil {
			return nil, err
		}
		bootBlock, err := blockcreator.BootstrapBlock(tx)
		if err != nil {
			return nil, err
		}
		if err = p.blockProcessor.Bootstrap(bootBlock); err != nil {
			return nil, err
		}
	}
```
I changed that to `ledgerHeight, err := conf.blockStore.Height()`.